src/meta/src/hummock/manager/mod.rs

```diff
@@ -1270,6 +1419,55 @@ where
             .await;
         }
 
+        // Warn of table_ids that is not found in expected compaction group.
+        // It indicates:
+        // 1. Either these table_ids are never registered to any compaction group. This is FATAL
+        // since compaction filter will remove these valid states incorrectly.
+        // 2. Or the owners of these table_ids have been dropped, but their stale states are still
+        // committed. This is OK since compaction filter will remove these stale states
+        // later.
+        let mut branch_sstables = vec![];
+        sstables.retain_mut(|(compaction_group_id, sst)| {
+            let res = match compaction_groups.get(compaction_group_id) {
+                Some(compaction_group) => {
+                    let mut is_valid = true;
+                    for table_id in sst
+                        .table_ids
+                        .iter()
+                        .filter(|t| !compaction_group.member_table_ids().contains(t))
+                    {
+                        is_valid = false;
+                        tracing::warn!(
+                            "table {} doesn't belong to expected compaction group {}",
+                            table_id,
+                            compaction_group_id
+                        );
+                    }
+                    is_valid
+                }
+                None => false,
+            };
+            if !res {
+                sst.divide_version += 1;
+                let mut branch_groups = HashSet::new();
+                for (group_id, group) in &compaction_groups {
```

Comment:
Actually we can only log instead of asserting when there is table_id in SST not found in any group because we cannot distinguish between table being dropped or table not assigned to group as mentioned in https://github.com/risingwavelabs/risingwave/pull/5823#discussion_r995446539

Reply:
> I think we need a way to check whether all table ids for a SST are assigned to a group and add corresponding assertion. IMO, it is easier to do so by iterating the table ids of the SST in the outer loop and find the corresponding compaction group in the inner loop. Just an example:
> 
> ```rust
>            let mut group_table_ids = HashMap::<u64, Vec<u32>>::new();
>                 for table_id in sst.get_table_ids() {
>                     match compaction_groups
>                         .iter()
>                         .find(|(_, group)| group.member_table_ids().contains(table_id))
>                     {
>                         Some((group_id, group)) => {
>                             group_table_ids
>                                 .entry(*group_id)
>                                 .or_default()
>                                 .push(*table_id);
>                         }
>                         None => {
>                             // Should be unreachable. Add assertion here.
>                             // TODO
>                         },
>                     }
>                 }
>                 if group_table_ids.len() == 1 {
>                     // No split. Just a group move
>                     // TODO
>                 } else {
>                     for (group_id, group) in group_table_ids {
>                         // A split group. Add a shadow SST.
>                         // TODO
>                     }
>                 }
> ```

fixed